rust/integration-tests/src/config.rs

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{anyhow, Context};
+use clap::Parser;
+use diesel::pg::PgConnection;
+use integration_tests::{
+    diff_test_helper::ProcessorTestHelper,
+    diff_tests::{all_tests::get_processor_map, remove_inserted_at, remove_transaction_timestamp},
+    DiffTest, TestType,
+};
+use processor::processors::token_v2_processor::TokenV2ProcessorConfig;
+use std::{fs, path::PathBuf, sync::Arc};
+// use aptos_indexer_test_transactions::{
+//     ALL_SCRIPTED_TRANSACTIONS, ALL_IMPORTED_MAINNET_TXNS, ALL_IMPORTED_TESTNET_TXNS,
+// };
+use testing_transactions::{
+    ALL_IMPORTED_MAINNET_TXNS, ALL_IMPORTED_TESTNET_TXNS, ALL_SCRIPTED_TRANSACTIONS,
+};
+
+const IMPORTED_TESTNET_FOLDER: &str = "expected_db_output_files/imported_testnet_txns";
+const IMPORTED_MAINNET_FOLDER: &str = "expected_db_output_files/imported_mainnet_txns";
+const SCRIPTED_FOLDER: &str = "expected_db_output_files/scripted_txns";
+// Constants for test context types
+const IMPORTED_TESTNET: &str = "imported_testnet";
+const IMPORTED_MAINNET: &str = "imported_mainnet";
+const SCRIPTED: &str = "scripted";
+
+#[derive(Parser)]
+#[clap(name = "Indexer db expected schema output CLI")]
+pub struct IndexerCliArgs {
+    /// List of test contexts to create, e.g., imported_testnet, mainnet, scripted
+    #[clap(long, num_args = 1.., required = true)]
+    pub test_contexts: Vec<String>,
+}
+
+impl IndexerCliArgs {
+    pub async fn run(&self) -> anyhow::Result<()> {
+        let processor_configs = vec![
+            TestProcessorConfig {
+                config: processor::processors::ProcessorConfig::EventsProcessor,
+            },
+            TestProcessorConfig {
+                config: processor::processors::ProcessorConfig::FungibleAssetProcessor,
+            },
+            TestProcessorConfig {
+                config: processor::processors::ProcessorConfig::TokenV2Processor(
+                    TokenV2ProcessorConfig {
+                        query_retries: 3,
+                        query_retry_delay_ms: 1000,
+                    },
+                ),
+            },
+        ];
```

Comment:
Would this work for custom processors?

Reply:
Yes, the current structure would support adding a new custom processor, but will need to do a few steps to integrate the new processor into the framework. 

assume you have a custom processor
1.  Integrate the Custom Processor by adding your custom processor to the processor_configs vector along with the other processors like this:
```
let processor_configs = vec![
    TestProcessorConfig {
        config: processor::processors::ProcessorConfig::EventsProcessor,
    },
    TestProcessorConfig {
        config: processor::processors::ProcessorConfig::FungibleAssetProcessor,
    },
    TestProcessorConfig {
        config: processor::processors::ProcessorConfig::TokenV2Processor(
            TokenV2ProcessorConfig {
                query_retries: 3,
                query_retry_delay_ms: 1000,
            },
        ),
    },
    TestProcessorConfig {
        config: processor::processors::ProcessorConfig::MyCustomProcessor(
            MyCustomProcessorConfig {
                query_retries: 3,
                query_retry_delay_ms: 500,
            },
        ),
    },
];
```
2. In the get_processor_map() function, ensure that your new custom processor is added to the processor map with the correct name:

```
pub fn get_processor_map() -> HashMap<String, Arc<Box<dyn ProcessorTestHelper>>> {
    let mut processor_map = HashMap::new();
    // Adding existing processors
    processor_map.insert(
        "TokenV2Processor".to_string(),
        Arc::new(Box::new(TokenV2ProcessorTestHelper)),
    );
    // Adding your new custom processor
    processor_map.insert(
        "MyCustomProcessor".to_string(),
        Arc::new(Box::new(MyCustomProcessorTestHelper)),
    );
    processor_map
}
```
3. When running the run function, the custom processor will create JSON output just like the other processors. The code handling this will already work for the new processor, as it will follow the pattern that we have for other processors like TokenV2Processor. The output folder and file structure will be automatically handled by the closure inside the run method.
and we don't need to modify the logic inside run as it is processor-agnostic.

i think there is a few steps I missed, but this should work with the custom processor as well

Reply:
Hmm.. is there a way to split out the config definition out from the tooling code? If the users have to update this vector, this model reverts back to the fork/clone model which SDK moves away from.

Reply:
agreed! let me see how I can transform this to SDK way!

Reply:
Synced offline: we want to provide an argument to cargo test so that they can decide to generate db output by passing in `cargo test -- --diff`, or run without `--diff` flag which is a default behavior of just running tests